language/return_spec.rb

```diff
@@ -249,4 +249,202 @@ def f
       ReturnSpecs::MethodWithBlock.new.method2.should == [0, 1, 2]
     end
   end
+
+  ruby_version_is '2.4' do
+    describe "at top level" do
+      it "stops file execution" do
+        ruby_exe(<<-END_OF_CODE).should == "before return\n"
+          puts "before return"
+          return
+
+          puts "after return"
+        END_OF_CODE
+
+        $?.exitstatus.should == 0
```

Comment:
`ruby_exe` is powerful but has the cost of spawning a new Ruby process every time.
This is particularly bad on some implementations where it can take anywhere from 10ms to a couple seconds.

I think those specs do not need `ruby_exe`, instead they could use a `tmp("top_return")` file (in a `before :each` block), `File.write()` the code in it and just call `Kernel#load`.
Then the puts should be replaced with `ScratchPad << "before return"` and `ScratchPad.recorded.should == ...`